### PR TITLE
Fix comparison of ConstraintWrapper

### PR DESCRIPTION
**Pull request by oheyzhiwei**
There was no proper `equals` in `ConstraintWrapper`, so foreign keys always generated a diff even if the underlying ddl didn't change. 

Added a new test case for this. I couldn't figure out how to make the `expectedDdlDiff` work with an empty line, so I gave up and made it generate `ALTER TABLE test1 DROP COLUMN col2`. Ideally, we should just check that there is no diff. Feel free to update this PR for a better test case

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM maven:3.8-openjdk-11
```
Comment (nielm) [inline Dockerfile:1]: Please remove this file from the PR.

File: TEST.md [hunk 2]
```diff
@@ -0,0 +1 @@
+## TESTPLZIGNORE
```
Comment (nielm) [inline TEST.md:1]: Please remove this file from the PR.

File: apply_diff.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (nielm) [inline apply_diff.sh:1]: Please remove this file from the PR.

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (nielm) [inline build.sh:1]: Please remove this file from the PR.

File: gen_diff.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (nielm) [inline gen_diff.sh:1]: Please remove this file from the PR.

**Comment by nielm** [comment]
Can you also split the deletion policy changes into a separate PR please

**Comment by zhiwei-arbo** [comment]
Apologies, looks like too many things changed since then and I accidentally pushed to the branch. Will look into splitting it up when I have time

**Comment by nielm** [comment]
> Apologies, looks like I accidentally pushed to this branch. Will look into splitting it up when I have time

No problem, and thanks for the contribution!

**Pull request closed without merging**